### PR TITLE
CLC-6459 Add Proxy to their course site as a Lead TA

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/CLC-6459

I've said this before, but this sprawling code desperately needs refactoring.

However, the refactor's not going to happen this week. (I've said that before, too.)

### CODE REVIEW

File: spec/models/canvas_csv/provide_course_site_spec.rb [hunk 1]
```diff
@@ -902,17 +939,41 @@
...
+        slug: 'chem-196',
+        title: 'Special Laboratory Study',
+        role: 'Instructor',
+        sections: [{
+          ccn: ccns[3],
+          courseCode: 'CHEM 196',
+          instruction_format: 'IND',
+          instructors: [
+            {uid: random_id, role: 'PI'},
+            {uid: adminstrative_proxy_id, role: 'APRX'}
```
Comment (paulkerschen) [inline spec/models/canvas_csv/provide_course_site_spec.rb]: @raydavis Here and at two places below, there's a typo `adminstrative`, which is luckily consistent with itself.

File: app/models/canvas_csv/provide_course_site.rb [hunk 2]
```diff
@@ -348,8 +364,10 @@ def generate_section_definitions(term_yr, term_cd, sis_course_id, campus_section
               'status' => 'active'
             }
             section_definitions << section_definition
-            if !@import_data['is_admin_by_ccns'] && section[:instructors].map{|instructor| instructor[:uid]}.include?(@uid)
+            if !@import_data['is_admin_by_ccns'] &&
+              (instructing_assignment = (section[:instructors].select{|instructor| instructor[:uid] == @uid}).first)
```
Comment (paulkerschen) [inline app/models/canvas_csv/provide_course_site.rb]: @raydavis FWIW, I tend to prefer `.find { |x| ... }` syntax to `.select { |x| ...}.first` as quicker and more readable. (Lines 148-149 also.)

**Comment by paulkerschen** [comment]
Looks as clear to me as could be reasonably hoped, @raydavis; I have slightly style-policed.

File: app/models/canvas_csv/provide_course_site.rb (hunk 2 above)
Comment (raydavis) [inline app/models/canvas_csv/provide_course_site.rb]: I too prefer it, when I can remember it.

**Comment by raydavis** [comment]
Thank you for walking down these mean streets, @paulkerschen !

**Pull request merged**